### DISCUSSION TITLE
Use onchain trade protocol for offchain trading

### DISCUSSION THREAD

**Discussion by rexi1e** [bisq-network/bisq#5756] [Ideas]
Bisq neds to move offchain, and some proposals are there i know.

I remember proposal from long time ago for bsq bonds and such, but if i recall those bonds were meant to serve as collateral for arbitrarily high trade amounts, which is why the idea failed to make it further.

But what if we use existing 2 of 2 protocol as base for offchain traddes between specific people? I am inspired by model of Lightning : do 1 on-chain tx to secure funds and create incentives for both people to hold up their side of deal over time. Bisq is already based on 2 of 2 multisig for security. But on bisq, instead of sending bitcoin through lightning to bypass blockchain, this could mean sending fiat/btc at specified intervals of time to bypass blockchain (e.g. sepa, zelle for fiat and lightning and liquid for btc). To take most advantage of this, should do bigger amounts over longer periods of time.

e.g.

Bob and Alice wanna trade 1 btc over 12 months. Bob is seller and Alice wants to buy. Assume there is new payment account type that is longer-term (perhaps defined in months) and that outlines a schedule of transactions (e.g. X btc every Y days). So Bob and Alice enter into a trade with these new payment accounts and open 2of2 multisig with 15% dep., so  0.15 btc each locked, and decide to exchange 0.042 btc every 2 weeks for 12 months. Bob sends btc over lightning and Alice pays with sepa.

[Mobile app like this](https://github.com/bisq-network/proposals/issues/342) would be super nice for reminders in this kind of senario.

I am not 100% sure of all details but it seems like the basic pieces are there to make this feasible.

Further thoughts:
- would be good to have some way to associate signed accounts to this new long-term payment method (e.g. SEPA --> SEPA Long-Term) so that users can benefit from the extra piece of mind provided by signing. Important if doing long-term trade relationship i think
- maybe it is better to have payment regime defined in fiat instead of btc; but not sure if this is avoidable since bisq trades are definedin btc
- thus ending the trade relationship needs more thoughts.
  - Closing trade channel after any of the mutual exchanges should be easy; but may need rules to disincentivze closing channels early in the relationship since it makes on-chain tx relatively expensive. 
  - Or Maybe there is some way to transfer obligation to another trusted trader (e.g. one with a signed account of same type)?

Dispute resolution also needs some thoughts. Im not sure if is possible to verify invoice generation and payments (?). Liquid could work for sure. Of course fiat side could be verified as usual.

I am long time user and really looking forard to seeing Bisq move ahead off the chain !!

**Comment by MwithM** [comment]
I I traded frequently with one user and I felt like we didn't need to release the deposit, as we would probably trade later in a month or couple months. I think that this is proposing something similar, but I don't find a huge advantage on it. 
If a lot of trust is established, there is no need Bisq, and if there's no much trust, or want to try a better offer somewhere else, traders won't like to let a deposit for a long time.
